zio-json/shared/src/main/scala-3/zio/json/macros.scala

```diff
@@ -42,6 +42,93 @@ final case class jsonDiscriminator(name: String) extends Annotation
 // does not provide a mechanism for obtaining the CaseClass associated to the
 // Subtype.
 
+sealed trait JsonMemberFormat extends (String => String)
+case class CustomCase(f: String => String) extends JsonMemberFormat {
+  override def apply(memberName: String): String = f(memberName)
+}
+case object SnakeCase extends JsonMemberFormat {
+  override def apply(memberName: String): String = jsonMemberNames.enforceSnakeOrKebabCase(memberName, '_')
+}
+case object CamelCase extends JsonMemberFormat {
+  override def apply(memberName: String): String =
+    jsonMemberNames.enforceCamelOrPascalCase(memberName, toPascal = false)
+}
+case object PascalCase extends JsonMemberFormat {
+  override def apply(memberName: String): String = jsonMemberNames.enforceCamelOrPascalCase(memberName, toPascal = true)
+}
+case object KebabCase extends JsonMemberFormat {
+  override def apply(memberName: String): String = jsonMemberNames.enforceSnakeOrKebabCase(memberName, '-')
+}
+
+/**
+ * If used on a case class, determines the strategy of member names
+ * transformation during serialization and deserialization. Four common
+ * strategies are provided above and a custom one to support specific use cases.
+ */
+final case class jsonMemberNames(format: JsonMemberFormat) extends Annotation
+private[json] object jsonMemberNames {
+
+  /**
+   * ~~Stolen~~ Borrowed from jsoniter-scala by Andriy Plokhotnyuk
```

Comment:
Thanks, @plokhotnyuk

Reply:
I’m happy to help!

BTW, please give your GitHub stars to [jsoniter-scala](https://github.com/plokhotnyuk/jsoniter-scala) or/and start sponsoring my GitHub account to support its development and adoption of its tricks here.